### PR TITLE
Move Pixel Bad FED Channel simulation related plugins to `CondTools`

### PR DESCRIPTION
**Pull request by mmusich**
#### PR description:

As per AlCa/DB request a bunch of Pixel Bad FED Channel simulation related plugins are moved from `CondFormats/SiPixelObjects` to `CondTools/SiPixel`.
I profit of the PR to fix a small bug in `FastSiPixelFEDChannelContainerFromQuality` and to apply code-formatting.
The moved plugins are now tested in unit tests.

#### PR validation:

Relies on existing unit tests.
Run `scram b runtests`

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

N/A

### CODE REVIEW

**Comment by mmusich** [comment]
@cmsbuild please test

**Comment by tvami** [comment]
+1
 * Jenkins tests pass -- expect for the known issues
 * unit tests pass https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-4a1593/21881/unitTests/src/CondTools/SiPixel/test/testPixelDBs/testing.log
 * changes in line w the PR describtion

File: CondTools/SiPixel/plugins/FastSiPixelFEDChannelContainerFromQuality.cc [hunk 1]
```diff
@@ -304,7 +306,7 @@ FastSiPixelFEDChannelContainerFromQuality::createFromSiPixelQuality(const SiPixe
     std::vector<PixelFEDChannel> disabledChannelsDetSet;
     std::vector<sipixelobjects::CablingPathToDetUnit> path = theFedCabling.pathToDetUnit(mod.DetID);
     unsigned int nrocs_inLink(0);
-    if (path.size() != 0) {
+    if (!path.empty()) {
```
Comment (perrotta) [inline CondTools/SiPixel/plugins/FastSiPixelFEDChannelContainerFromQuality.cc:309]: This check is useless, if independently of it you later divide by `nrocs_inLink` anyhow.
You can remove it, or (maybe better) throw some exception inside

File: CondTools/SiPixel/plugins/SiPixelFEDChannelContainerFromQualityConverter.cc [hunk 2]
```diff
@@ -135,7 +135,7 @@ SiPixelFEDChannelContainerFromQualityConverter::createFromSiPixelQuality(const S
     std::vector<sipixelobjects::CablingPathToDetUnit> path = theFedCabling.pathToDetUnit(mod.DetID);
     auto cabling_ = theFedCabling.cablingTree();
     unsigned int nrocs_inLink(0);
-    if (path.size() != 0) {
+    if (!path.empty()) {
```
Comment (perrotta) [inline CondTools/SiPixel/plugins/SiPixelFEDChannelContainerFromQualityConverter.cc:138]: This check is useless, if independently of it you later divide by `nrocs_inLink` anyhow.
You can remove it, or (maybe better) throw some exception inside

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by tvami** [comment]
+1
 * see https://github.com/cms-sw/cmssw/pull/36762#issuecomment-1018947510

**Comment by perrotta** [comment]
+1

**Comment by perrotta** [comment]
merge

**Pull request merged**